### PR TITLE
WIP: Update to new securesystemslib API

### PR DESCRIPTION
**Pull request by jku**
### Update to new (currently unreleased) securesystemslib API
    
  * API changes covered:
    * keys and interface modules removed
    * SSlibSigner removed
    * CryptoSigner added: this replaces the removed functionality
    * DSSE "signatures" container type changed
  * Currently pins a securesystemslib main branch commit:
    this should be reverted before merging, when securesystemslib
    has made a release
  * tests/generated_data/generate_md.py was simplified
  * Encrypted test keys in tests/repository_data/keystore were replaced
    with the unencrypted PEM versions of the same keys
  * The public test keys in tests/repository_data/keystore were removed
    as they were not used anymore

It's a bit annoyingly in a single commit... but I was working on this fixing one test failure at a time: reasonable commits were not really an option during the work. Anyway, I think this proves the securesystemslib API changes are solid enough.

TODO:
* [x] wait for securesystemslib release
* [x] revert the pinned.txt change, update the dpendency in pyproject.toml

### CODE REVIEW

**Comment by jku** [comment]
tests fail because of the macos upgrade (#2618), passes locally

**Comment by jku** [comment]
rebase on main

File: examples/uploader/_localrepo.py [hunk 1]
```diff
@@ -115,19 +123,24 @@ def add_target(self, role: str, targetpath: str) -> bool:
...
-        with open(f"{self.key_dir}/{role}", "w", encoding="utf-8") as f:
-            f.write(json.dumps(keydict))
+        with open(f"{self.key_dir}/{role}", "wb") as f:
+            # TODO this is dumb and  needs to be securesystemslibs job...
+            priv_key = signer._private_key.private_bytes(
+                encoding=Encoding.PEM,
+                format=PrivateFormat.PKCS8,
+                encryption_algorithm=NoEncryption(),
+            )
+            f.write(priv_key)
```
Comment (lukpueh) [inline examples/uploader/_localrepo.py]: cc https://github.com/secure-systems-lab/securesystemslib/pull/799

File: tests/test_api.py [hunk 2]
```diff
@@ -86,13 +80,17 @@ def setUpClass(cls) -> None:
             os.path.join(test_repo_data, "keystore"), cls.keystore_dir
         )
 
-        # Load keys into memory
-        cls.keystore = {}
-        for role in ["delegation", Snapshot.type, Targets.type, Timestamp.type]:
```
Comment (lukpueh) [inline tests/test_api.py:91]: Was the "delegation" key unused?

**Review by lukpueh — COMMENTED** [review]
Great stuff!

File: tests/test_api.py (hunk 2 above)
Comment (jku) [reply tests/test_api.py:91]: yes

File: examples/uploader/_localrepo.py (hunk 1 above)
Comment (jku) [reply examples/uploader/_localrepo.py]: replaced with `signer.private_bytes`

**Comment by jku** [comment]
I suppose this is now ready for review :shrug:

**Review by lukpueh — APPROVED** [review]
Looks good. There's one mention of SSlibSigner left in a docstring. I can remove it.

https://github.com/theupdateframework/python-tuf/blob/38f309bbbf817c0f06de29b1cec85c0b14feb0a4/tuf/api/metadata.py#L356

**Pull request merged**